src/main/java/com/rybka/view/ExchangeView.java

```diff
@@ -34,18 +37,10 @@ public void showDialog() {
             System.out.print("Insert target currency: ");
             var userTargetCurrency = scanner.next().toUpperCase();
 
-            var currencyResponse = service.loadCurrencyOf(userBaseCurrency, userTargetCurrency);
-
-            var total = service.calculateTotal(currencyResponse.getRate(), userValue);
-            var convertedResult = constructConvertedResult(currencyResponse.getBase(),
-                    currencyResponse.getTarget(),
-                    userValue,
-                    currencyResponse.getRate(),
-                    total);
-
-            currencyHistoryDAO.save(convertedResult);
-
-            showExchange(convertedResult);
+            var convertTask = new ConvertTask(service, currencyHistoryDAO, userBaseCurrency, userTargetCurrency, userValue);
```

Comment:
Each time showDialog() method called we will build new ConvertCommand and execute it. Do you think this is correct? 
Would be better to define service and reuse it each time when command called?